src/BlazorIdentity.UI/LoginWith2fa.razor

```diff
@@ -0,0 +1,107 @@
+﻿@typeparam TUser where TUser : class
+
+@using Microsoft.Extensions.Logging
+@inject IBlazorSignInManager<TUser> SignInManager
+@inject IBlazorUserManager<TUser> UserManager
+@inject ILogger<LoginWith2fa<TUser>> logger
+@inject NavigationManager NavManager
+
+<hr />
+<p>Your login is protected with an authenticator app. Enter your authenticator code below.</p>
+
+    <EditForm class="col-5" EditContext="editContext" OnSubmit="@HandleSubmit">
+        <DataAnnotationsValidator />
+        <ValidationSummary />
+        <input type="hidden" id="rememberMe" @bind-value="@RememberMe" />
+
+        <div class="mb-3">
+            <label for="TwoFactorCode" class="form-label">Authenticator code</label>
+            <InputText id="TwoFactorCode" class="form-control" @bind-Value="loginModel.TwoFactorCode" />
+        </div>
+
+        <div class="mb-3 form-check">
+            <InputCheckbox id="rememberMe" class="form-check-input" @bind-Value="loginModel.RememberMachine" />
+            <label class="form-check-label" for="exampleCheck1">Remember this machine</label>
+        </div>
+        <div>
+            <button type="submit" class="btn btn-primary">Log in</button>
+        </div>
+    </EditForm>
+<p>
+    Don't have access to your authenticator device? You can
+    <a id="recovery-code-login" href="/Account/LoginWithRecoveryCode">log in with a recovery code</a>.
```

Comment:
I'm going to have think a bit more about this navigation model as having the component just blindly link to a URL that it doesn't itself carry a page for feels wrong. I don't think Blazor page components have the same capability as Razor Pages whereby a Razor library can bring in pages that are routable that can then be easily overridden by the hosting app so I don't think the same model will work here.